ui/helpers/utils/icon-factory.js

```diff
@@ -18,11 +19,7 @@ function IconFactory(jazzicon) {
 }
 
 IconFactory.prototype.iconForAddress = function (address, diameter) {
-  let addr = address;
-
-  if (isHexString(addHexPrefix(address))) {
-    addr = checksumAddress(addHexPrefix(address));
-  }
+  const addr = toChecksumHexAddress(address);
```

Comment:
I was about to comment that this was a change in behaviour, because the hex prefix always gets added. But apparently we were doing this in v9.4 as well! So this is good as-is.